### DISCUSSION TITLE
AIdeck onboard control crazyflie

### DISCUSSION THREAD

**Discussion by Sheng-Qian-LI** [bitcraze/discussions#1275] [Q&A]
Hi.

We want to implement autonomous flight for the Crazyflie using on-board. After image processing ,determine the flight direction.

Is it possible to control the flight direction of the Crazyflie only using the AI deck without connecting to PC (via python)?

Thank you.

**Comment by gemenerik** [comment]
Yes, this is possible with [CPX](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/cpx/). Take a look at [this example](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/other-examples/stm_gap8_cpx/).

**Reply by cuber-1** [reply]
What exactly does the example you provided do?

**Reply by ToveRumar** [reply]
This example shows how to communicate back and forth between the Crazyflie 2.1 main cpu and the GAP8 using CPX. 

You can send basic direction commands using this API; https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/crtp/crtp_commander/ 
If you want to use a higher level commander you can look at this documentation for how the commander structure works; https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/sensor-to-control/commanders_setpoints/

I do think the latency of CPX might be to high to control the flight directions, directly. But using the high level commander and sending set points over CPX should work well.  

I hope this answered your question!
